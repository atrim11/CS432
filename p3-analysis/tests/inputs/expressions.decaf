def int main() {
    void result;
    result = ((2*7-5/(3+8)%7)/5 - 9 % 2 / 3) / 2 + 7 - 3 - 3 - 3 - 3 - 5 * 9 + 0 <= 2;
    result = ((a + b * (c - d / e) && f || g % h == i) != (j <= k && l >= m) || (n < o && p > q) + ((r == s) && (t != u)) - ((v * w / x) % y));
    result = ((5 + foo(3, bar(2)) * (arr[0] - arr[1] / 2) && baz(4)) || (x <= 7 && y >= 8) != (z < 10));
    result = (((10 == 15) && (a[2] + b[3]) || (c / 5) != d) - (e * f) + g % (8 - h) + (i && j));
    result = (((alpha() * beta(3, 5) - (gamma(4) + delta[x])) < (5 == 5) || (m >= n && p < q)));
    result = (((5 + (foo() - bar[1])) * (3 == (4 + 1))) >= (6 % 2)) || (7 - 8) != (9 < (a[0] * b[1]));
    result = ((funcA(1, 2) + (funcB(3) * (x + y) / z) % 4) && (5 != 6) || (array[0] < array[1] && array[2] > 10));
    result = ((x && (y + z) - (funcC(a[0], b[1]) * (c[2] / d[3]))) != (10 > 12) || (foo[4] < bar[5]));
    result = (((1 + 2) * (3 - (4 / 5)) % (arr[3] && arr[4])) == (foo(6, 7) || bar(8, 9)));
    result = ((x + y * (10 - (z / 2))) >= (3 * arr[0] - funcD(4)) && (5 != 6));
    result = ((foo(a) && bar(b, c)) || (baz(d) >= (x + y) && (e <= f)));
    result = (((1 + (2 * (3 - 4))) % (5 / 6)) == (7 != (8 && 9))) || (foo(arr[0]) && (bar(arr[1], 10)));
    result = ((alpha + beta * (gamma - delta) / (epsilon % z) && a[0]) || (b[1] != c[2]));
    result = ((funcA(x) + (arr[0] / arr[1]) >= (5 - funcB(y))) && (10 == 20));
    result = (((x * y) - (foo(2, 3) % (4 + bar[1]))) <= (5 || (6 > 7)));
    result = (((a < b) && (c >= d)) || ((foo(1) - bar(2)) < (3 + 4)));
    result = (((5 + (foo(a) * b[1])) / (10 - c[2])) < (7 && (8 || 9)));
    result = (((a[1] * b[2]) >= (foo(3, 4) && bar(5))) || (x < y));
    result = (((10 % (5 + 3)) * (4 / 2)) == (foo(a) || bar(b)));
    result = (((arr[0] + arr[1]) - (arr[2] * arr[3])) < (5 / 6)) && (funcC(x) || funcD(y));
    result = ((1 + (2 - (3 * (4 / 5))) % (6 && 7)) == (foo(8) < bar(9)));
    result = ((x && (y + z) - (funcC(a[0], b[1]) * (c[2] / d[3]))) != (10 > 12) || (foo[4] < bar[5]));

    x = x + 5;
    x = foo(10) * 2;
    y = 10;
    arr[3] = arr[2] + 1;
    b = isReady && isLoaded;
    processData(x);
    y = x + foo(y) - 3;
    arr[i] = getValue();
    result = a + b * 2;
    foo(bar(3, 4), 5);
    return 0;
}