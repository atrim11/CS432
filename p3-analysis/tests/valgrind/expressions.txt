==194664== Memcheck, a memory error detector
==194664== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==194664== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==194664== Command: ../decaf inputs/expressions.decaf
==194664== 
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
ERROR: No 'type' attribute
Invalid: Variable 'result' declared as void on line 2
Type mismatch in assignment on line 3: expected void, got ???
Error: Variable 'a' used without being defined on line 4
Error: Variable 'b' used without being defined on line 4
Error: Variable 'c' used without being defined on line 4
Error: Variable 'd' used without being defined on line 4
Error: Variable 'e' used without being defined on line 4
Error: Variable 'f' used without being defined on line 4
Error: Variable 'g' used without being defined on line 4
Error: Variable 'h' used without being defined on line 4
Error: Variable 'i' used without being defined on line 4
Error: Variable 'j' used without being defined on line 4
Error: Variable 'k' used without being defined on line 4
Error: Variable 'l' used without being defined on line 4
Error: Variable 'm' used without being defined on line 4
Error: Variable 'n' used without being defined on line 4
Error: Variable 'o' used without being defined on line 4
Error: Variable 'p' used without being defined on line 4
Error: Variable 'q' used without being defined on line 4
Error: Variable 'r' used without being defined on line 4
Error: Variable 's' used without being defined on line 4
Error: Variable 't' used without being defined on line 4
Error: Variable 'u' used without being defined on line 4
Error: Variable 'v' used without being defined on line 4
Error: Variable 'w' used without being defined on line 4
Error: Variable 'x' used without being defined on line 4
Error: Variable 'y' used without being defined on line 4
Type mismatch in assignment on line 4: expected void, got ???
Symbol 'foo' undefined on line 5
Symbol 'bar' undefined on line 5
Error: Variable 'arr' used without being defined on line 5
Error: Variable 'arr' used without being defined on line 5
Symbol 'baz' undefined on line 5
Error: Variable 'x' used without being defined on line 5
Error: Variable 'y' used without being defined on line 5
Error: Variable 'z' used without being defined on line 5
Type mismatch in assignment on line 5: expected void, got ???
Error: Variable 'a' used without being defined on line 6
Error: Variable 'b' used without being defined on line 6
Error: Variable 'c' used without being defined on line 6
Error: Variable 'd' used without being defined on line 6
Error: Variable 'e' used without being defined on line 6
Error: Variable 'f' used without being defined on line 6
Error: Variable 'g' used without being defined on line 6
Error: Variable 'h' used without being defined on line 6
Error: Variable 'i' used without being defined on line 6
Error: Variable 'j' used without being defined on line 6
Type mismatch in assignment on line 6: expected void, got ???
Symbol 'alpha' undefined on line 7
Symbol 'beta' undefined on line 7
Symbol 'gamma' undefined on line 7
Error: Variable 'delta' used without being defined on line 7
Error: Variable 'x' used without being defined on line 7
Error: Variable 'm' used without being defined on line 7
Error: Variable 'n' used without being defined on line 7
Error: Variable 'p' used without being defined on line 7
Error: Variable 'q' used without being defined on line 7
Type mismatch in assignment on line 7: expected void, got ???
Symbol 'foo' undefined on line 8
Error: Variable 'bar' used without being defined on line 8
Error: Variable 'a' used without being defined on line 8
Error: Variable 'b' used without being defined on line 8
Type mismatch in assignment on line 8: expected void, got ???
Symbol 'funcA' undefined on line 9
Symbol 'funcB' undefined on line 9
Error: Variable 'x' used without being defined on line 9
Error: Variable 'y' used without being defined on line 9
Error: Variable 'z' used without being defined on line 9
Error: Variable 'array' used without being defined on line 9
Error: Variable 'array' used without being defined on line 9
Error: Variable 'array' used without being defined on line 9
Type mismatch in assignment on line 9: expected void, got ???
Error: Variable 'x' used without being defined on line 10
Error: Variable 'y' used without being defined on line 10
Error: Variable 'z' used without being defined on line 10
Symbol 'funcC' undefined on line 10
Error: Variable 'a' used without being defined on line 10
Error: Variable 'b' used without being defined on line 10
Error: Variable 'c' used without being defined on line 10
Error: Variable 'd' used without being defined on line 10
Error: Variable 'foo' used without being defined on line 10
Error: Variable 'bar' used without being defined on line 10
Type mismatch in assignment on line 10: expected void, got ???
Error: Variable 'arr' used without being defined on line 11
Error: Variable 'arr' used without being defined on line 11
Symbol 'foo' undefined on line 11
Symbol 'bar' undefined on line 11
Type mismatch in assignment on line 11: expected void, got ???
Error: Variable 'x' used without being defined on line 12
Error: Variable 'y' used without being defined on line 12
Error: Variable 'z' used without being defined on line 12
Error: Variable 'arr' used without being defined on line 12
Symbol 'funcD' undefined on line 12
Type mismatch in assignment on line 12: expected void, got ???
Symbol 'foo' undefined on line 13
Error: Variable 'a' used without being defined on line 13
Symbol 'bar' undefined on line 13
Error: Variable 'b' used without being defined on line 13
Error: Variable 'c' used without being defined on line 13
Symbol 'baz' undefined on line 13
Error: Variable 'd' used without being defined on line 13
Error: Variable 'x' used without being defined on line 13
Error: Variable 'y' used without being defined on line 13
Error: Variable 'e' used without being defined on line 13
Error: Variable 'f' used without being defined on line 13
Type mismatch in assignment on line 13: expected void, got ???
Symbol 'foo' undefined on line 14
Error: Variable 'arr' used without being defined on line 14
Symbol 'bar' undefined on line 14
Error: Variable 'arr' used without being defined on line 14
Type mismatch in assignment on line 14: expected void, got ???
Error: Variable 'alpha' used without being defined on line 15
Error: Variable 'beta' used without being defined on line 15
Error: Variable 'gamma' used without being defined on line 15
Error: Variable 'delta' used without being defined on line 15
Error: Variable 'epsilon' used without being defined on line 15
Errordot: graph is too large for cairo-renderer bitmaps. Scaling by 0.977623 to fit
: Variable 'z' used without being defined on line 15
Error: Variable 'a' used without being defined on line 15
Error: Variable 'b' used without being defined on line 15
Error: Variable 'c' used without being defined on line 15
Type mismatch in assignment on line 15: expected void, got ???
Symbol 'funcA' undefined on line 16
Error: Variable 'x' used without being defined on line 16
Error: Variable 'arr' used without being defined on line 16
Error: Variable 'arr' used without being defined on line 16
Symbol 'funcB' undefined on line 16
Error: Variable 'y' used without being defined on line 16
Type mismatch in assignment on line 16: expected void, got ???
Error: Variable 'x' used without being defined on line 17
Error: Variable 'y' used without being defined on line 17
Symbol 'foo' undefined on line 17
Error: Variable 'bar' used without being defined on line 17
Type mismatch in assignment on line 17: expected void, got ???
Error: Variable 'a' used without being defined on line 18
Error: Variable 'b' used without being defined on line 18
Error: Variable 'c' used without being defined on line 18
Error: Variable 'd' used without being defined on line 18
Symbol 'foo' undefined on line 18
Symbol 'bar' undefined on line 18
Type mismatch in assignment on line 18: expected void, got ???
Symbol 'foo' undefined on line 19
Error: Variable 'a' used without being defined on line 19
Error: Variable 'b' used without being defined on line 19
Error: Variable 'c' used without being defined on line 19
Type mismatch in assignment on line 19: expected void, got ???
Error: Variable 'a' used without being defined on line 20
Error: Variable 'b' used without being defined on line 20
Symbol 'foo' undefined on line 20
Symbol 'bar' undefined on line 20
Error: Variable 'x' used without being defined on line 20
Error: Variable 'y' used without being defined on line 20
Type mismatch in assignment on line 20: expected void, got ???
Symbol 'foo' undefined on line 21
Error: Variable 'a' used without being defined on line 21
Symbol 'bar' undefined on line 21
Error: Variable 'b' used without being defined on line 21
Type mismatch in assignment on line 21: expected void, got ???
Error: Variable 'arr' used without being defined on line 22
Error: Variable 'arr' used without being defined on line 22
Error: Variable 'arr' used without being defined on line 22
Error: Variable 'arr' used without being defined on line 22
Symbol 'funcC' undefined on line 22
Error: Variable 'x' used without being defined on line 22
Symbol 'funcD' undefined on line 22
Error: Variable 'y' used without being defined on line 22
Type mismatch in assignment on line 22: expected void, got ???
Symbol 'foo' undefined on line 23
Symbol 'bar' undefined on line 23
Type mismatch in assignment on line 23: expected void, got ???
Error: Variable 'x' used without being defined on line 24
Error: Variable 'y' used without being defined on line 24
Error: Variable 'z' used without being defined on line 24
Symbol 'funcC' undefined on line 24
Error: Variable 'a' used without being defined on line 24
Error: Variable 'b' used without being defined on line 24
Error: Variable 'c' used without being defined on line 24
Error: Variable 'd' used without being defined on line 24
Error: Variable 'foo' used without being defined on line 24
Error: Variable 'bar' used without being defined on line 24
Type mismatch in assignment on line 24: expected void, got ???
Error: Variable 'x' used without being defined on line 26
Error: Variable 'x' used without being defined on line 26
Error: Variable 'x' used without being defined on line 27
Symbol 'foo' undefined on line 27
Error: Variable 'y' used without being defined on line 28
Type mismatch in assignment on line 28: expected ???, got int
Error: Variable 'arr' used without being defined on line 29
Error: Variable 'arr' used without being defined on line 29
Error: Variable 'b' used without being defined on line 30
Error: Variable 'isReady' used without being defined on line 30
Error: Variable 'isLoaded' used without being defined on line 30
Symbol 'processData' undefined on line 31
Error: Variable 'x' used without being defined on line 31
Error: Variable 'y' used without being defined on line 32
Error: Variable 'x' used without being defined on line 32
Symbol 'foo' undefined on line 32
Error: Variable 'y' used without being defined on line 32
Error: Variable 'arr' used without being defined on line 33
Error: Variable 'i' used without being defined on line 33
Symbol 'getValue' undefined on line 33
Error: Variable 'a' used without being defined on line 34
Error: Variable 'b' used without being defined on line 34
Type mismatch in assignment on line 34: expected void, got ???
Symbol 'foo' undefined on line 35
Symbol 'bar' undefined on line 35
==194664== 
==194664== HEAP SUMMARY:
==194664==     in use at exit: 0 bytes in 0 blocks
==194664==   total heap usage: 6,188 allocs, 6,188 frees, 896,496 bytes allocated
==194664== 
==194664== All heap blocks were freed -- no leaks are possible
==194664== 
==194664== For lists of detected and suppressed errors, rerun with: -s
==194664== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
