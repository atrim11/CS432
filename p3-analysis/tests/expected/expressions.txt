Void variable 'result' on line 2
Type mismatch: void is incompatible with bool on line 3
Symbol 'a' undefined on line 4
Symbol 'b' undefined on line 4
Symbol 'c' undefined on line 4
Symbol 'd' undefined on line 4
Symbol 'e' undefined on line 4
Symbol 'f' undefined on line 4
Type mismatch: bool expected but int found on line 4
Symbol 'g' undefined on line 4
Symbol 'h' undefined on line 4
Symbol 'i' undefined on line 4
Symbol 'j' undefined on line 4
Symbol 'k' undefined on line 4
Symbol 'l' undefined on line 4
Symbol 'm' undefined on line 4
Symbol 'n' undefined on line 4
Symbol 'o' undefined on line 4
Symbol 'p' undefined on line 4
Symbol 'q' undefined on line 4
Symbol 'r' undefined on line 4
Symbol 's' undefined on line 4
Symbol 't' undefined on line 4
Symbol 'u' undefined on line 4
Type mismatch: int expected but bool found on line 4
Type mismatch: int expected but bool found on line 4
Symbol 'v' undefined on line 4
Symbol 'w' undefined on line 4
Symbol 'x' undefined on line 4
Symbol 'y' undefined on line 4
Type mismatch: bool expected but int found on line 4
Type mismatch: void is incompatible with bool on line 4
Symbol 'foo' undefined on line 5
Symbol 'bar' undefined on line 5
Symbol 'arr' undefined on line 5
Symbol 'arr' undefined on line 5
Symbol 'baz' undefined on line 5
Type mismatch: bool expected but int found on line 5
Symbol 'x' undefined on line 5
Symbol 'y' undefined on line 5
Symbol 'z' undefined on line 5
Type mismatch: void is incompatible with bool on line 5
Symbol 'a' undefined on line 6
Symbol 'b' undefined on line 6
Type mismatch: bool expected but int found on line 6
Symbol 'c' undefined on line 6
Symbol 'd' undefined on line 6
Symbol 'e' undefined on line 6
Symbol 'f' undefined on line 6
Type mismatch: int expected but bool found on line 6
Symbol 'g' undefined on line 6
Symbol 'h' undefined on line 6
Symbol 'i' undefined on line 6
Symbol 'j' undefined on line 6
Type mismatch: int expected but bool found on line 6
Type mismatch: void is incompatible with int on line 6
Symbol 'alpha' undefined on line 7
Symbol 'beta' undefined on line 7
Symbol 'gamma' undefined on line 7
Symbol 'delta' undefined on line 7
Symbol 'x' undefined on line 7
Type mismatch: int expected but bool found on line 7
Symbol 'm' undefined on line 7
Symbol 'n' undefined on line 7
Symbol 'p' undefined on line 7
Symbol 'q' undefined on line 7
Type mismatch: void is incompatible with bool on line 7
Symbol 'foo' undefined on line 8
Symbol 'bar' undefined on line 8
Type mismatch: int expected but bool found on line 8
Symbol 'a' undefined on line 8
Symbol 'b' undefined on line 8
Type mismatch: int is incompatible with bool on line 8
Type mismatch: void is incompatible with bool on line 8
Symbol 'funcA' undefined on line 9
Symbol 'funcB' undefined on line 9
Symbol 'x' undefined on line 9
Symbol 'y' undefined on line 9
Symbol 'z' undefined on line 9
Type mismatch: bool expected but int found on line 9
Symbol 'array' undefined on line 9
Symbol 'array' undefined on line 9
Symbol 'array' undefined on line 9
Type mismatch: void is incompatible with bool on line 9
Symbol 'x' undefined on line 10
Symbol 'y' undefined on line 10
Symbol 'z' undefined on line 10
Symbol 'funcC' undefined on line 10
Symbol 'a' undefined on line 10
Symbol 'b' undefined on line 10
Symbol 'c' undefined on line 10
Symbol 'd' undefined on line 10
Type mismatch: bool expected but int found on line 10
Symbol 'foo' undefined on line 10
Symbol 'bar' undefined on line 10
Type mismatch: void is incompatible with bool on line 10
Symbol 'arr' undefined on line 11
Symbol 'arr' undefined on line 11
Type mismatch: int expected but bool found on line 11
Symbol 'foo' undefined on line 11
Symbol 'bar' undefined on line 11
Type mismatch: int is incompatible with bool on line 11
Type mismatch: void is incompatible with bool on line 11
Symbol 'x' undefined on line 12
Symbol 'y' undefined on line 12
Symbol 'z' undefined on line 12
Symbol 'arr' undefined on line 12
Symbol 'funcD' undefined on line 12
Type mismatch: void is incompatible with bool on line 12
Symbol 'foo' undefined on line 13
Symbol 'a' undefined on line 13
Symbol 'bar' undefined on line 13
Symbol 'b' undefined on line 13
Symbol 'c' undefined on line 13
Symbol 'baz' undefined on line 13
Symbol 'd' undefined on line 13
Symbol 'x' undefined on line 13
Symbol 'y' undefined on line 13
Symbol 'e' undefined on line 13
Symbol 'f' undefined on line 13
Type mismatch: void is incompatible with bool on line 13
Type mismatch: bool expected but int found on line 14
Type mismatch: bool expected but int found on line 14
Type mismatch: int is incompatible with bool on line 14
Type mismatch: int is incompatible with bool on line 14
Symbol 'foo' undefined on line 14
Symbol 'arr' undefined on line 14
Symbol 'bar' undefined on line 14
Symbol 'arr' undefined on line 14
Type mismatch: void is incompatible with bool on line 14
Symbol 'alpha' undefined on line 15
Symbol 'beta' undefined on line 15
Symbol 'gamma' undefined on line 15
Symbol 'delta' undefined on line 15
Symbol 'epsilon' undefined on line 15
Symbol 'z' undefined on line 15
Symbol 'a' undefined on line 15
Type mismatch: bool expected but int found on line 15
Symbol 'b' undefined on line 15
Symbol 'c' undefined on line 15
Type mismatch: void is incompatible with bool on line 15
Symbol 'funcA' undefined on line 16
Symbol 'x' undefined on line 16
Symbol 'arr' undefined on line 16
Symbol 'arr' undefined on line 16
Symbol 'funcB' undefined on line 16
Symbol 'y' undefined on line 16
Type mismatch: void is incompatible with bool on line 16
Symbol 'x' undefined on line 17
Symbol 'y' undefined on line 17
Symbol 'foo' undefined on line 17
Symbol 'bar' undefined on line 17
Type mismatch: bool expected but int found on line 17
Type mismatch: int expected but bool found on line 17
Type mismatch: void is incompatible with bool on line 17
Symbol 'a' undefined on line 18
Symbol 'b' undefined on line 18
Symbol 'c' undefined on line 18
Symbol 'd' undefined on line 18
Symbol 'foo' undefined on line 18
Symbol 'bar' undefined on line 18
Type mismatch: void is incompatible with bool on line 18
Symbol 'foo' undefined on line 19
Symbol 'a' undefined on line 19
Symbol 'b' undefined on line 19
Symbol 'c' undefined on line 19
Type mismatch: bool expected but int found on line 19
Type mismatch: bool expected but int found on line 19
Type mismatch: bool expected but int found on line 19
Type mismatch: int expected but bool found on line 19
Type mismatch: void is incompatible with bool on line 19
Symbol 'a' undefined on line 20
Symbol 'b' undefined on line 20
Symbol 'foo' undefined on line 20
Symbol 'bar' undefined on line 20
Type mismatch: int expected but bool found on line 20
Symbol 'x' undefined on line 20
Symbol 'y' undefined on line 20
Type mismatch: void is incompatible with bool on line 20
Symbol 'foo' undefined on line 21
Symbol 'a' undefined on line 21
Symbol 'bar' undefined on line 21
Symbol 'b' undefined on line 21
Type mismatch: int is incompatible with bool on line 21
Type mismatch: void is incompatible with bool on line 21
Symbol 'arr' undefined on line 22
Symbol 'arr' undefined on line 22
Symbol 'arr' undefined on line 22
Symbol 'arr' undefined on line 22
Symbol 'funcC' undefined on line 22
Symbol 'x' undefined on line 22
Symbol 'funcD' undefined on line 22
Symbol 'y' undefined on line 22
Type mismatch: void is incompatible with bool on line 22
Type mismatch: bool expected but int found on line 23
Type mismatch: bool expected but int found on line 23
Type mismatch: int expected but bool found on line 23
Symbol 'foo' undefined on line 23
Symbol 'bar' undefined on line 23
Type mismatch: int is incompatible with bool on line 23
Type mismatch: void is incompatible with bool on line 23
Symbol 'x' undefined on line 24
Symbol 'y' undefined on line 24
Symbol 'z' undefined on line 24
Symbol 'funcC' undefined on line 24
Symbol 'a' undefined on line 24
Symbol 'b' undefined on line 24
Symbol 'c' undefined on line 24
Symbol 'd' undefined on line 24
Type mismatch: bool expected but int found on line 24
Symbol 'foo' undefined on line 24
Symbol 'bar' undefined on line 24
Type mismatch: void is incompatible with bool on line 24
Symbol 'x' undefined on line 26
Symbol 'x' undefined on line 26
Symbol 'x' undefined on line 27
Symbol 'foo' undefined on line 27
Symbol 'y' undefined on line 28
Symbol 'arr' undefined on line 29
Symbol 'arr' undefined on line 29
Symbol 'b' undefined on line 30
Symbol 'isReady' undefined on line 30
Symbol 'isLoaded' undefined on line 30
Symbol 'processData' undefined on line 31
Symbol 'x' undefined on line 31
Symbol 'y' undefined on line 32
Symbol 'x' undefined on line 32
Symbol 'foo' undefined on line 32
Symbol 'y' undefined on line 32
Symbol 'arr' undefined on line 33
Symbol 'i' undefined on line 33
Symbol 'getValue' undefined on line 33
Symbol 'a' undefined on line 34
Symbol 'b' undefined on line 34
Type mismatch: void is incompatible with int on line 34
Symbol 'foo' undefined on line 35
Symbol 'bar' undefined on line 35
